File: jgrpc/src/main/java/jgrpc/lib/sym/Variable.java
package jgrpc.lib.sym;

public class Variable extends Symbol {
    public Type type;

    public Variable(String n, Type t, String s, Location l) {
        super(n, s, l);
        type = t;
    }

    @Override
    public SymType getType() {
        return SymType.VAR;
    }
}

File: grpc/src/main/scala/grpc/lib/symbol/Variable.scala
package grpc
package lib
package symbol

import grpc.lib.symbol.Type.Type

class Variable(val name: String, private val typ: Type, val scope: String,
               val location: Location)
  extends Symbol(name, scope, location) {

  def getType: Type = typ

  override def getSymType = SymType.VAR
}
